helm/kubernetes-nginx-ingress-controller-chart/templates/controller-hpa.yaml

```diff
@@ -0,0 +1,26 @@
+apiVersion: autoscaling/v2beta1
+kind: HorizontalPodAutoscaler
+metadata:
+  name: {{ .Values.controller.name }}
+  namespace: {{ .Values.namespace }}
+  labels:
+    app: {{ .Values.controller.name }}
+    giantswarm.io/service-type: "managed"
+    k8s-app: {{ .Values.controller.k8sAppLabel }}
+spec:
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: {{ .Values.controller.name }}
+  minReplicas: 2
+  # This is the maximum number of nodes we allow currently. Having more controllers than nodes should not happen.
+  maxReplicas: 100
```

Comment:
To be hyper safe. We should probably set this lower, but I don't know of a sensible value.

Reply:
not sure if this should be a maxed out with number of nodes as @calvix suggested, or maybe a multiple of max number of nodes? but for now I guess let's use 100 and we should somehow test this